File: samples/js/rag/text_embeddings.js
import { TextEmbeddingPipeline, PoolingType } from 'openvino-genai-node';
import { basename } from 'node:path';

main();

async function main() {
    const modelPath = process.argv[2];
    const texts = process.argv.slice(3);

    const usageCommand = `Usage: node ${basename(process.argv[1])} <MODEL_DIR> '<TEXT 1>' ['<TEXT 2>' ...]`;
    if (!modelPath) {
        console.error('Please specify path to model directory');
        console.error(usageCommand);
        process.exit(1);
    }
    if (!texts.length) {
        console.error('Please specify prompt');
        console.error(usageCommand);
        process.exit(1);
    }

    const device = 'CPU'; // GPU can be used as well
    const config = {
        'pooling_type': PoolingType.MEAN
    };

    const pipeline = await TextEmbeddingPipeline(modelPath, device, config);

    await pipeline.embedDocuments(texts);
}

File: samples/python/rag/text_embeddings.py
import argparse
import openvino_genai


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("model_dir")
    args = parser.parse_args()

    device = "CPU"  # GPU can be used as well

    config = openvino_genai.TextEmbeddingPipeline.Config()
    config.pad_to_max_length = True
    config.batch_size = 4
    config.max_length = 128
    config.pooling_type = openvino_genai.TextEmbeddingPipeline.PoolingType.MEAN

    pipeline = openvino_genai.TextEmbeddingPipeline(args.model_dir, device, config)

    text_embeddings = pipeline.embed_documents(args.texts)
    query_embeddings = pipeline.embed_query("What is the capital of France?")


if "__main__" == __name__:
    main()
